scheduler/src/cook/rest/api.clj

```diff
@@ -2121,10 +2158,28 @@
                          [true {::error (str "Must supply at least one job or group to start."
                                              "Are you specifying that this is application/json?")}]
 
-                         (and pool-name (not pool))
+                         :let [db (db conn)
```

Comment:
I think a comment may be worth it here. "We reject jobs unless either the pool exists or the pool is a job-routing-pool."

And we can simplify this code, I think as:

(or (not pool-name) (job-routing-pool-name? pool-name) (pool-name->exists?-cache ....)

With a comment with pool-name->exists?-cache saying that the cached value is either true or false. (Oooh! you are storing true or false here,  right? Caches cannot store nil values or keys.)

Reply:
Adding a comment, good call.

The reason I'm creating a variable for:

```
skip-pool-name-checks?
(or (not pool-name)
    (job-routing-pool-name? pool-name))
```

is that I re-use that same `skip-pool-name-checks?` condition in the next check (the one for whether or not the pool is accepting new job submissions).

The caches are always caching `true` or `false` (the result of either `some?` or `pool/accepts-submissions?`).

Reply:
Added comment